pages/create-a-tool.tsx

```diff
@@ -0,0 +1,496 @@
+    ModalHeader,
+    ModalFooter,
+    useDisclosure,
+    useToast,
+} from "@chakra-ui/react";
+import { SelfCheckQuestionCard, ToolHomePage } from "@components";
+
+//Self Check Questions React functional component
+const Home: React.FC = () => {
+    //self check tool
+    const defaultTool = [
+        {
+            _id: "50e642d7e4a1ae34207a92a0",
+            title: "",
+            type: "",
+            thumbnail: "",
+            video: "",
+            description: "",
+            linkedModule: "",
+            relatedResources: [
+                ["", ""],
+                ["", ""],
+                ["", ""],
+            ],
+            relatedStories: [
+                ["", ""],
+                ["", ""],
+                ["", ""],
+            ],
+            externalResources: [
+                ["", ""],
+                ["", ""],
+                ["", ""],
+            ],
+            recommendedTools: ["", "", ""],
+        },
+    ];
+    const [toolList, setToolList] = useState(defaultTool);
+    const [lastSavedTool, setLastSavedTool] = useState(
+        JSON.parse(JSON.stringify(defaultTool)),
+    );
+
+    //
+
+    const changeInput = (id, target, type, index1 = null, index2 = null) => {
+        const newTool = toolList.slice(0);
+        console.log(target, type, index1, index2, [
+            newTool.findIndex((e) => e._id === id),
+        ]);
+        if (index2 != null) {
+            newTool[newTool.findIndex((e) => e._id === id)][type][index1][
+                index2
+            ] = target;
+        } else if (index1 != null) {
+            newTool[newTool.findIndex((e) => e._id === id)][type][index1] =
+                target;
+        } else {
+            newTool[newTool.findIndex((e) => e._id === id)][type] = target;
+        }
+        setToolList(newTool);
+    };
+
+    const thumbnailValidation = (id, target) => {
+        const newTool = toolList.slice(0);
+        const fileType = target.split(".").pop().toLowerCase();
+        const allowedFileTypes = ["jpg", "jpeg", "png", "gif", "bmp", "gif"];
+        if (allowedFileTypes.includes(fileType)) {
+            newTool[newTool.findIndex((e) => e._id === id)].thumbnail = target;
+            setToolList(newTool);
+            toast({
+                title: "Success",
+                description: "Your thumbnail has been uploaded",
+                status: "success",
+                position: "bottom-left",
+                duration: 5000,
+                isClosable: true,
+            });
+        } else {
+            if (target != "") {
+                toast({
+                    title: "Invalid file type",
+                    description: "Please upload a jpg, jpeg, bmp, gif, or png",
+                    status: "error",
+                    position: "bottom-left",
+                    duration: 5000,
+                    isClosable: true,
+                });
+                document.getElementById("thumbnail").value = "";
```

Comment:
pass the thumbnail to `ToolHomePage` as a prop and set the value in `Input` line 92 in ToolHomePage

Reply:
Changed to what we discussed.